RearrangingCarsTest.java

```diff
@@ -90,6 +92,32 @@ public void testDifferentLengthInputs() {
         Assert.assertNull(moves);
     }
 
```

Comment:
I think more tests should be added. We don't just want to test whether the number of moves we get is indeed optimal, but also that the added code is working. I can add comments about possible testing in the above implementation and we can iterate again.